subsys/bluetooth/controller/hci/hci.c

```diff
@@ -5253,6 +5271,91 @@ static void le_ext_adv_report(struct pdu_data *pdu_data,
 	adv_info->length = data_len;
 	memcpy(&adv_info->data[0], data, data_len);
 
+	if (!scan_data) {
+		node_rx_extra_list_release(node_rx->hdr.rx_ftr.extra);
+
+		return;
+	}
+
+	/* Allocate, append as buf fragement and construct the scan response
+	 * event.
+	 */
```

Comment:
As per my previous comment: It seems like the following addition of scan response handling could be put into a separate function.

Reply:
Refactored in to `ext_adv_info_fill()`